--- worker-html.js	2020-04-30 08:49:58.000000000 -0500
+++ worker-html-planstone.js	2020-04-30 09:13:34.000000000 -0500
@@ -2692,6 +2704,12 @@
 			tokenizer.setState(tag_name_state);
 		} else if (data === '!') {
 			tokenizer.setState(markup_declaration_open_state);
+		} else if (data === '$') {//MCC
+			tokenizer._currentToken = {type: 'Comment', data: ''};
+			tokenizer.setState(comment_start_state);
+		} else if (data === '#') {//MCC
+			tokenizer._currentToken = {type: 'Comment', data: ''};
+			tokenizer.setState(comment_start_state);
 		} else if (data === '/') {
 			tokenizer.setState(close_tag_open_state);
 		} else if (data === '>') {
@@ -3128,6 +3146,9 @@
 			tokenizer.setState(data_state);
 		} else if (data === '-') {
 			tokenizer.setState(comment_end_dash_state);
+		} else if (data === '>') { //MCC
+			buffer.unget(data);
+			tokenizer.setState(comment_end_state);
 		} else if (data === '\u0000') {
 			tokenizer._parseError("invalid-codepoint");
 			tokenizer._currentToken.data += "\uFFFD";
@@ -3179,8 +3200,8 @@
 			tokenizer._currentToken.data += "--\uFFFD";
 			tokenizer.setState(comment_state);
 		} else {
-			tokenizer._parseError("unexpected-char-in-comment");
-			tokenizer._currentToken.data += '--' + data;
+            //MCC tokenizer._parseError("unexpected-char-in-comment");
+            //MCC tokenizer._currentToken.data += '--' + data;
 			tokenizer.setState(comment_state);
 		}
 		return true;
